src/nvim/path.c

```diff
@@ -400,6 +400,33 @@ char_u *save_absolute_path(const char_u *name)
   return vim_strsave((char_u *) name);
 }
 
+/// Checks if a path has a wildcard character including '~', unless at the end.
+/// @param p  The path to expand.
+/// @returns Unix: True if it contains one of "?[{`'$".
+/// @returns Windows: True if it contains one of "*?$[".
+bool path_has_wildcard(const char_u *p)
+  FUNC_ATTR_NONNULL_ALL
+{
+  for (; *p; mb_ptr_adv(p)) {
+#if defined(UNIX)
+    if (p[0] == '\\' && p[1] != NUL) {
+      p++;
+      continue;
+    }
+
+    const char *wildcards = "*?[{`'$";
+#else
+    // Windows:
+    // TODO(SplinterOfChaos): Assuming we compile with VIM_BACKTICK.
```

Comment:
hmm... probably not. I don't think I looked up what `VIM_BACKTICK` did before, but I don't imagine we won't support it. (Shell expansion for certain situations.)

Removed the TODO.